aiostream/stream/combine.py

```diff
@@ -76,14 +78,22 @@ async def zip(*sources: AsyncIterable[T]) -> AsyncIterator[tuple[T, ...]]:
             await stack.enter_async_context(streamcontext(source)) for source in sources
         ]
         # Loop over items
+        STOP_SENTINEL = object()  # only used in strict mode
```

Comment:
It might be easier to type if `STOP_SENTINEL` is an enum sentinel:
```suggestion
        _StopSentinelType = enum.Enum("_StopSentinelType", "STOP_SENTINEL")
        STOP_SENTINEL = _StopSentinelType.STOP_SENTINEL
```